### PR TITLE
Match log

### PR DESCRIPTION
**Pull request by alenmonti**
## Summary of changes

- Add improvements for the mobile experience, such as closing the sidebar when a match is found or the user clicks to log in, rules, and leaderboard.
- Add notifications when receiving a friend request or a challenge
- add notifications indicating the opponent´s action (Truco, envido, quiero, etc)
- Add custom notifications on win or lose
- On wide screens chat starts open
- Explicit match log-in chat was not added because of not fit well with the game flow

New library added, you may need to run npm install.
Pull match log branch and test notifications and mobile experience for the review.

### CODE REVIEW

File: truco-front/src/gameLogic/GameActionsManager.ts [hunk 1]
```diff
@@ -65,6 +66,9 @@ export default class GameActionsManager {
     }
 
     public handleAction(actionMessage: GameActionMessage, iCalled: boolean = false) {
+        if(!iCalled && actionMessage.action !== GameAction.PLACE_CARD && actionMessage.action !== GameAction.NONE){
+            toast(getPrintableAction(actionMessage.action))
```
Comment (MelnikAndres) [inline truco-front/src/gameLogic/GameActionsManager.ts:70]: This line of code is part of the view, this logic shouldn't be here.

File: truco-front/src/gameLogic/type/GameAction.ts [hunk 2]
```diff
@@ -47,4 +47,21 @@ export const isEnvidoAction = (action: GameAction): boolean => {
 
 export const isTurnAction = (action: GameAction): boolean => {
     return action === GameAction.PLACE_CARD || action === GameAction.IR_AL_MAZO
+}
+
+export const getPrintableAction = (action: GameAction): string => {
```
Comment (MelnikAndres) [inline truco-front/src/gameLogic/type/GameAction.ts:52]: i like this, maybe we can add a few variations to every response

File: truco-front/src/hooks/usePusherListeners.ts [hunk 3]
```diff
@@ -153,10 +155,12 @@ export const usePusherListeners = (
         })
 
         gameManager.events.addOnFriendRequestListener((friendUser: User) => {
+            toast(friendUser.username + " sent you a friend request!")
```
Comment (MelnikAndres) [inline truco-front/src/hooks/usePusherListeners.ts:158]: after this, you can find the "onFriendRequestAcceptedListener" that triggers when someone accepts a friend request, maybe you can use toast there, too.

File: truco-front/src/hooks/usePusherListeners.ts [hunk 4]
```diff
@@ -153,10 +155,12 @@ export const usePusherListeners = (
         })
 
         gameManager.events.addOnFriendRequestListener((friendUser: User) => {
+            toast(friendUser.username + " sent you a friend request!")
             setFriendRequests((prev) => [...prev, friendUser])
         })
 
         gameManager.events.addOnGameChallengeListener((challenger: string) => {
+            toast(challenger + " challenged you to a game!")
             gameManager.acceptChallenge(challenger) // TODO replace with logic
```
Comment (MelnikAndres) [inline truco-front/src/hooks/usePusherListeners.ts:164]: Maybe you want to do it on a later update, but here we need a way to accept or deny the challenge

**Review by MelnikAndres — CHANGES_REQUESTED** [review]

**Comment by MelnikAndres** [comment]
You can also update the challenge button

File: truco-front/src/hooks/usePusherListeners.ts (hunk 4 above)
Comment (alenmonti) [reply truco-front/src/hooks/usePusherListeners.ts:164]: I think it's way outside the scope of this issue

File: truco-front/src/hooks/usePusherListeners.ts (hunk 3 above)
Comment (alenmonti) [reply truco-front/src/hooks/usePusherListeners.ts:158]: Added

File: truco-front/src/gameLogic/type/GameAction.ts (hunk 2 above)
Comment (alenmonti) [reply truco-front/src/gameLogic/type/GameAction.ts:52]: Done

File: truco-front/src/gameLogic/GameActionsManager.ts (hunk 1 above)
Comment (alenmonti) [reply truco-front/src/gameLogic/GameActionsManager.ts:70]: ready

**Review by MelnikAndres — APPROVED** [review]

**Pull request merged**